Index: dbt7/DSGen-software-code-4.0.0_final/tools/tokenizer.l
===================================================================
--- dbt7.orig/DSGen-software-code-4.0.0_final/tools/tokenizer.l
+++ dbt7/DSGen-software-code-4.0.0_final/tools/tokenizer.l
@@ -90,6 +90,7 @@ int yylook(void);
 %{ 
 /* macro definitions go here */ 
 %}
+%option noinput nounput
 INT	-?[0-9]+
 DECIMAL   -?[0-9]*\.[0-9]+
 WS	[ \t]+
